calypso/calypso.go

```diff
@@ -0,0 +1,264 @@
+package calypso
+
+import (
+	"bytes"
+
+	"go.dedis.ch/dela/crypto"
+	"go.dedis.ch/dela/serde"
+
+	"go.dedis.ch/dela/calypso/json"
+	"go.dedis.ch/dela/calypso/storage"
+	"go.dedis.ch/dela/calypso/storage/inmemory"
+	"go.dedis.ch/dela/dkg"
+	"go.dedis.ch/dela/ledger/arc"
+	"go.dedis.ch/dela/ledger/arc/darc"
+	serdej "go.dedis.ch/dela/serde/json"
+	"go.dedis.ch/kyber/v3"
+	"go.dedis.ch/kyber/v3/suites"
+	"golang.org/x/xerrors"
+)
+
+const (
+	// ArcRuleUpdate defines the rule to update the arc. This rule must be set
+	// at the write creation to allow the arc to be latter updated.
+	ArcRuleUpdate = "calypso_update"
+	// ArcRuleRead defines the arc rule to read a value
+	ArcRuleRead = "calypso_read"
+)
+
+// Suite is the Kyber suite for Pedersen.
+var suite = suites.MustFind("Ed25519")
+
+// NewCalypso creates a new Calypso
+func NewCalypso(dkg dkg.DKG) *Caly {
+	return &Caly{
+		dkg:     dkg,
+		storage: inmemory.NewInMemory(),
+	}
+}
+
+// Caly is a wrapper around DKG to provide a storage and authorization layer
+//
+// implements calypso.Secret
+type Caly struct {
+	dkg      dkg.DKG
+	dkgActor dkg.Actor
+	storage  storage.KeyValue
+}
+
+// Listen implements calypso.Secret
+func (c *Caly) Listen() error {
+	if c.dkgActor != nil {
+		return xerrors.Errorf("listen has already been called")
+	}
+
+	actor, err := c.dkg.Listen()
+	if err != nil {
+		return xerrors.Errorf("failed to listen dkg: %v", err)
+	}
+
+	c.dkgActor = actor
```

Comment:
Sorry I just noticed that you can make your life easier by asking a DKG actor only. The one instantiating will take care of calling the `Listen` function. That's part of the idea of splitting the setup and actor logic.

Reply:
Then you can remove the point to the structure.

Reply:
not sure what you mean by "asking a DKG actor only"..?

Reply:
Calypso can use the actor only, so instead of having New(dkg), you can have New(actor).